examples/common/python/connectors/direct/tcs_listener/tcs_worker_registry_handler.py

```diff
@@ -73,88 +91,51 @@ def __worker_registry_handler_on_boot(self):
-
-        if 'workerId' in input_json_str:
-            worker_id = str(input_json['params']['workerId'])
-        else:
-            return utility.create_error_response(
-                WorkerError.INVALID_PARAMETER_FORMAT_OR_VALUE,
-                input_json["id"],
-                "Worker Id not found in the database. \
-                     Hence invalid parameter")
-
-        if(input_json['method'] == "WorkerRegister"):
-            return self.__process_worker_register(worker_id, input_json, response)
-        elif(input_json['method'] == "WorkerSetStatus"):
-            return self.__process_worker_set_status(worker_id, input_json, response)
-        elif(input_json['method'] == "WorkerRetrieve"):
-            return self.__process_worker_retrieve(worker_id, response)
-        elif(input_json['method'] == "WorkerUpdate"):
-            return self.__process_worker_update(worker_id, input_json, response)
 # ------------------------------------------------------------------------------------------------
-
-    def __process_worker_register(self, worker_id, input_value_json, response):
+    def WorkerRegister(self, **params):
         """
         Function to register a new worker to the enclave
         Parameters:
             - worker_id is the worker id specified in the input json str.
-            - input_json_str is a worker request json as per TCF API 5.3.1 Worker Register JSON Payload.
+            - input_json_str is a worker request json as per TCF API
+                5.3.1 Worker Register JSON Payload.
             - response is the response object to be returned to client.
         """
 
-        jrpc_id = input_value_json["id"]
+        worker_id = must_get_worker_id(params)
 
-        if(self.kv_helper.get("workers", worker_id) is None):
-            input_value = {}
-            input_value = input_value_json['params']
-            # Worker Initial Status is set to Active
-            input_value["status"] = WorkerStatus.ACTIVE
-
-            input_json_str = json.dumps(input_value)
-            self.kv_helper.set("workers", worker_id, input_json_str)
-            response = utility.create_error_response(
-                WorkerError.SUCCESS, jrpc_id,
-                "Successfully Registered")
-        else:
-            response = utility.create_error_response(
+        if(self.kv_helper.get("workers", worker_id) is not None):
+            raise JSONRPCDispatchException(
                 WorkerError.INVALID_PARAMETER_FORMAT_OR_VALUE,
-                jrpc_id,
-                "Worker Id already exists in the database. \
-                 Hence invalid parameter")
-        return response
+                "Worker Id already exists in the database." +
+                " Hence invalid parameter")
+
+        # Worker Initial Status is set to Active
+        params["status"] = WorkerStatus.ACTIVE
+
+        input_json_str = json.dumps(params)
+        self.kv_helper.set("workers", worker_id, input_json_str)
```

Comment:
worker register and worker set status should return functionality not implemented, since there is no functionality in enclave manager side to start the new worker and set the status to other possible values.

Reply:
Sorry, I don't got it = =. Would you mind showing me  the reference codes ~ @Ram-srini

Reply:
WorkerRegister() and WorkerSetStatus() functionalities are not implemented in the back end(in Enclave manager). For registering new worker should start a new enclave and setting worker status to OFFLINE, COMMISSIONED, etc.. is not happening in the background. Currently only one worker is supported and it is being started by enclave manager during boot time. Time being we should return with new ENUM  FUNCTIONALITY_NOT_IMPLEMENTED until we implement those APIs.

Reply:
@Ram-srini This we should probably handle in another PR. @sammyne has just used existing error codes and error messages as this PR is related to replacing existing implementation with JSON-JRPC library for handling request.

Reply:
Thanks for clarification~ @Ram-srini @manojgop 😄